### ISSUE TITLE
getting error while running BuildAndRunSimulation

### ISSUE THREAD

**Issue by georgethomas08** [digitaltwinconsortium/ManufacturingDTDLOntologies#6]
Hi team,

I'm getting this error while running BuildAndRunSimulation

![image](https://user-images.githubusercontent.com/8665537/177648684-3d204a66-5277-4fb1-a686-ec320498cab8.png)

from Docker desktop I can find all the related containers

![image](https://user-images.githubusercontent.com/8665537/177648902-10a7010c-1d9f-45a3-abc4-d76d41d615a3.png)

**Comment by barnstee** [comment]
Looks like you ran BuildAndRunSimulation twice, instead of calling StopSimulation in between?

**Comment by georgethomas08** [comment]
Hi @barnstee 

I have ran only once, if I delete all the containers and Run the file will it resolve? or any other solution to resolve this?

**Comment by barnstee** [comment]
Stop simulation will clean up everything, so yes. Then simply run BuildAndRunSimulation afterwards.

**Comment by georgethomas08** [comment]
Hi @barnstee ,

I found the issue there were two network with same name each, i deleted all the networks and rebuilt it, 

now except these containers all other are working, these containers are keep on restarting

![image](https://user-images.githubusercontent.com/8665537/177730955-f4bf74f2-0e33-4304-aab7-beec3d18625e.png)

**Comment by barnstee** [comment]
Check the logs of ua-cloudcommander and also verify that you have replaced all the occurrences of ENTER_IOT_HUB_HOSTNAME_HERE, ENTER_UACOMMANDER_DEVICE_NAME_HERE and ENTER_UACOMMANDER_DEVICE_KEY_HERE in the BuildAndRunSimulation.cmd file.

**Comment by georgethomas08** [comment]
Hi @barnstee 

Thanks for the support, I made mistake instead of primary key I used primary connection string, now everything is running but unable to monitor the telemetry using Azure IoT Explorer


this is the error I'm getting port 8081 is in use 
![image](https://user-images.githubusercontent.com/8665537/177756184-a6524b3f-1d43-4de3-a4b4-579b9896661a.png)


I can see one of the containers already using this

![image](https://user-images.githubusercontent.com/8665537/177756480-abb1cbf0-82e4-4c6f-b87e-c6878790692c.png)



I have tried using the VS code monitor the built-in endpoint but no data is getting displayed , but from Azure metric I can see messages are getting populated

![image](https://user-images.githubusercontent.com/8665537/177756976-36d17cc8-1f1a-4c6b-8b4b-c4dc832c5a4e.png)

There are my queries regarding this

1. How to view the telemetry data?
2. How get the view of the digital twin model and twin graph, do need to setup it manually?
3. Is it mandatory to store the data in ADX to get the complete view?

**Comment by barnstee** [comment]
You can simply change the port of the publisher in question in the following file:
https://github.com/digitaltwinconsortium/ManufacturingDTDLOntologies/blob/aa1d5e9f7f880b02cf68d1cf6d6f10c032040727/FactorySimulation/BuildAndRunSimulation.cmd#L92

Regarding visualization, I recommend ADX dashboards. ADX can be configured using this writeup:
https://www.linkedin.com/pulse/using-azure-data-explorer-opc-ua-erich-barnstedt/
Just ignore the bits in relation to configuring the publisher as you have already done that.

There are also sample ADX visualization queries here:
https://github.com/digitaltwinconsortium/ManufacturingDTDLOntologies/blob/main/ADXQueries/SampleDashboardQuery.kql

**Comment by georgethomas08** [comment]
Thanks @barnstee 

Coming to digital twin, we need to setup those twin models and model graph, I can see two DTDL files in the folder, do you have any documentation how to setup ?

**Comment by barnstee** [comment]
Use Azure Digital Twin Explorer to upload them. You can also upload the entire graph in one go.

**Comment by barnstee** [comment]
closing since issue was resolved.
